File: python/aula08/Matriz.py
from Dado import Dado
def iniciar_matriz(qtdL, qtdC):
    return[[0 for _ in range(qtdC)] for _ in range(qtdL)]
    pass

def exibir_matriz(matriz, qtdL, qtdC):
    for i in range(qtdL):
        for j in range(qtdC):
            print(matriz, end="\t")
            
    print()
    
    
def converter_matriz_lista(matriz, lista):
    for i, linha in enumerate(matriz):                
        for j, valor in enumerate(linha):
            if valor != 0:                
                lista.append(Dado(valor, i, j))
                
def exibir_lista(lista):
    for item in lista:
        print(item)

File: java/aula08/Matriz.java
java.Aula08
import java.util.ArrayList;

public class Matriz{
    public static void iniciarMatriz(int matriz[][], int qtdL, int qtdC){
        for(int i = 0; i < qtdL; i++){
            for(int j= 0; j < qtdC; j++){
                matriz[i][j] = 0;
            }
        }
    }

public static void exibirMatriz(int matriz[][], int qtdL, int qtdC){
    for (int i = 0; i < qtdL; i++) {
        for (int j = 0; j < qtdC; j++) {
            System.out.print(matriz[i][j] + "\t");
        }
        System.out.println();
    }     
}

public static void converterMatrizLista(int matriz[][], int qtdL, int qtdC, ArrayList<Dado> lista){
    for(int i = 0; i < qtdL; i++){
            for(int j= 0; j < qtdC; j++){
                if(matriz[i][j] != 0){
                    lista.add(new Dado(matriz[i][j], i,j));
                }
            }
        }
    }

    public static void exibirLista(ArrayList<Dado> lista){
        for(Dado item : lista){
            System.out.println(item);
        }
        System.out.println("--------------------------");
        System.out.println("Total de elementos da lista: " + lista.size());
    }
}